server/service/src/processors/transfer/shipment/create_inbound_shipment.rs

```diff
@@ -66,6 +67,18 @@ impl ShipmentTransferProcessor for CreateInboundShipmentProcessor {
         if linked_shipment.is_some() {
             return Ok(None);
         }
+        // 5.
+        let store = StoreRowRepository::new(connection)
+            .find_one_by_id(&record_for_processing.other_party_store_id)?
+            .ok_or(RepositoryError::NotFound)?;
+
+        if let Some(created_date) = store.created_date {
+            let store_created_datetime = NaiveDateTime::new(created_date, NaiveTime::from_hms_opt(0, 0, 0).unwrap_or_default());
+            let invoice_created_datetime = outbound_shipment.invoice_row.created_datetime - Duration::days(30);
```

Comment:
When I saw this earlier I wasn't sure but now writing a late comment - shouldn't you subtract the duration from the `store_created_datetime`? 

```
store_created_datetime = 2024-03-12
invoice_created_datetime = 2024-03-20
invoice_created_datetime = invoice_created_datetime - 30days // ~2024-02-20

invoice_created_date < store_created_datetime == True // We're not making transfers for invoices after the store was created? The goal was only make invoices if `store_created_date - invoice_created_date > -30days`, not `store_created_date - invoice_created_date > +30days`

```

It's late and I might be cooked though but I didn't want to get this browser tab lost again lol

Reply:
You are right!